osu.Game/Overlays/Music/MusicActionHandler.cs

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using osu.Framework.Allocation;
+using osu.Framework.Bindables;
+using osu.Framework.Graphics;
+using osu.Framework.Input.Bindings;
+using osu.Game.Beatmaps;
+using osu.Game.Input.Bindings;
+using osu.Game.Overlays.OSD;
+
+namespace osu.Game.Overlays.Music
+{
+    /// <summary>
+    /// Handles <see cref="GlobalAction"/> relating to music playback, and displays a <see cref="Toast"/> via the cached <see cref="OnScreenDisplay"/> accordingly.
+    /// </summary>
+    public class MusicActionHandler : Component, IKeyBindingHandler<GlobalAction>
+    {
+        [Resolved]
+        private IBindable<WorkingBeatmap> beatmap { get; set; }
+
+        [Resolved]
+        private MusicController musicController { get; set; }
+
+        [Resolved(canBeNull: true)]
+        private OnScreenDisplay onScreenDisplay { get; set; }
+
+        public bool OnPressed(GlobalAction action)
+        {
+            if (beatmap.Disabled)
+                return false;
+
+            switch (action)
+            {
+                case GlobalAction.MusicPlay:
+                    if (musicController.TogglePause())
+                        onScreenDisplay?.Display(new MusicActionToast(musicController.IsPlaying ? "Play track" : "Pause track"));
+
+                    return true;
+
+                case GlobalAction.MusicNext:
+                    musicController.NextTrack(() =>
+                    {
+                        onScreenDisplay?.Display(new MusicActionToast("Next track"));
+                    }).RunTask();
```

Comment:
we cannot have this returned task running business. i have *no idea* why this is calling `RunTask`, but it can't. there's no logical reasoning and this code reads abhorrently.

Reply:
did it mostly just to not do any change in behaviour, I agree that it looks bad though, will keep it scheduled then.